examples/Receiver_Response.py

```diff
@@ -0,0 +1,245 @@
+    thisfile = file2, nodes_in_kHz=False
+)
+# {{{ Calculate input V (acquired on Oscilloscope)
+for j, nodename in enumerate(control_nodenames):
+    d = psd.find_file(
+        file1,
+        expno=nodename,
+        exp_type=data_dir,
+    )
+    # {{{ fit signal in t domain to complex exponential
+    A, omega, phi, t = symbols("A omega phi t", real=True)
+    f = psd.lmfitdata(d)
+    f.functional_form = A * sp.exp(
+        1j * 2 * pi * omega * t + 1j * phi
+    )
+    f.set_guess(
+        A=dict(value=5e-3, min=1e-4, max=1),
+        omega=dict(
+            value=control["nu_test"][j],
+            min=control["nu_test"][j] - 1e4,
+            max=control["nu_test"][j] + 1e4,
+        ),
+        phi=dict(value=0.75, min=-pi, max=pi),
+    )
+    f.fit(use_jacobian=False)
+    f.eval()
+    # }}}
+    V_amp = f.output("A")
+    control["nu_test", j] = abs(V_amp) * 1e6  # μV
+# {{{ make spline for power going into RX box
+control.rename(
+    "nu_test", Dnu_name
+)  # since we will be applying $\Delta\nu$ axis to spline
+Pin_spline = control.spline_lambda()
+# }}}
+# }}}
+# {{{ Calculate dg
+with psd.figlist_var() as fl:
+    for j, nodename in enumerate(rec_nodenames):
+        d = psd.find_file(
+            file2,
+            exp_type=data_dir,
+            expno=nodename,
+        )
+        carrier = (
+            d.get_prop("acq_params")["carrierFreq_MHz"]
+            * 1e6
+        )  # $\nu_{RX,LO}$
+        rec_data["nu_test", j] = d
+        SW = str(
+            d.get_prop("acq_params")["SW_kHz"]
+        )  # For labeling final plot
+    rec_data.rename(
+        "nScans", "capture"
+    )  # To be more consistent with the oscilloscope data rename the nScans
+    #    dimension
+    acq_time = np.diff(rec_data["t"][r_[0, -1]]).item()
+    # {{{ Calculate PSD for each frequency (we will calculate power from the A
+    #     of the convolved test signal)
+    rec_data.run(
+        np.conj
+    )  # Empirically needed to give offset that increases with
+    #                        field
+    rec_data.ft("t", shift=True)  # $dg\sqrt{s/Hz}$
+    rec_data = abs(rec_data) ** 2  # $dg^{2}*s/Hz$
+    rec_data.mean("capture")
+    rec_data /= acq_time  # $dg^{2}/Hz$
+    # Convolve using $\lambda_{G}$ specified above
+    rec_data.convolve(
+        "t", lambda_G, enforce_causality=False
+    )
+    # }}}
+    # {{{ Calculate power of test signal (Eq. S3)
+    rec_data *= (
+        lambda_G / (2 * np.sqrt(np.log(2)))
+    ) * np.sqrt(pi)
+    rec_data.run(np.sqrt)  # dg
+    # center data at 0 MHz thus converting to $\Delta\nu$ rather than
+    # $\nu_{test}$
+    rec_data["nu_test"] = rec_data["nu_test"] - carrier
+    rec_data.rename("nu_test", Dnu_name)
+    rec_data.rename("t", nu_direct)
+    # {{{ Plot 2D pcolor
+    fig = plt.figure()
+    fl.next("2D plot", fig=fig)
+    the_2D = rec_data.C.run(np.real).pcolor(
+        cmap="BuPu", fig=fig, scale_independently=True
+    )
```

Comment:
I manually played with the function pcolor to get the bar and labels on top for the paper fig. But here the positioning doesn't seem important for the color bar

Reply:
I agree